addons/mail/static/src/messaging/entity/thread/thread.js

```diff
@@ -427,7 +434,7 @@ function ThreadFactory({ Entity }) {
                     },
                 });
             }
-            this.unlink({ renamingDiscuss: null });
+            this.env.messaging.discuss.unlink({ renamingThreads: this });
```

Comment:
I feel like the inverse here was actually more correct. Not good to introduce more access to singleton.